test/functional/fetchers_test.rb

```diff
@@ -54,4 +54,52 @@ def assert_fetcher_failed_cleanly(run_result, error_regex, profile_location)
     end
 
   end
+
+  # Refs #4727
+  describe "when a archive is available of an unfetchable profile with --airgap" do
+
+    let(:invocation) { "archive #{fetcher_profiles}/#{profile_name} --airgap" }
+
+    def teardown
+      FileUtils.rm_rf "#{fetcher_profiles}/#{profile_name}/vendor"
+      FileUtils.rm_f "#{profile_name}-0.1.0.tar.gz"
+    end
+
+    def assert_archive_worked(run_result)
+      _(run_result.stderr).must_be_empty
+      _(run_result.stdout).must_include "Finished archive generation"
+      assert_exit_code(0, run_result)
+    end
```

Comment:
Please use `prepare_examples` or the `tmpdir` option in `run_inspec_process`.

Reply:
Agreed, I'll aim to rework the tests to use tmpdir option to `run_inspec_process`.